content/blog/analysis-with-RCrunch.Rmd

```diff
@@ -0,0 +1,300 @@
+```{r}
+respondent_col <- as.vector(so_data$Respondent)
+length(respondent_col)
+```
+
+But we don't actually need to pull all of the data from a column just to know how many rows there are, we can use `nrow` with the dataset object, and the Crunch package is smart enough to use the metadata it already has to tell us how many rows there are.
+```{r}
+nrow(so_data)
+```
+
+The time savings are not trivial either, pulling the data:
+```{r}
+clearCache() # let's make this a fair fight
+system.time({
+    respondent_col <- as.vector(so_data$Respondent)
+    length(respondent_col)  
+})
+
+```
+
+Using the Crunch package's built-in tools, we see a speed up of approaching 5 times.
+```{r}
+clearCache() # let's make this a fair fight
+system.time({nrow(so_data)})
+```
+
+## Summary statistics
+Dimensions aren't the only time we can use Crunch to help us speed up our analyses. Crunch can also calculate some summary statistics on the server rather than transferring all of the data, and then running them locally. Let's look at the `min`, `mean`, `sd`, and `max` of the `JobSatisfaction` variable.
+
+```{r}
+min(so_data$JobSatisfaction, na.rm = TRUE)
+mean(so_data$JobSatisfaction, na.rm = TRUE)
+sd(so_data$JobSatisfaction, na.rm = TRUE)
+max(so_data$JobSatisfaction, na.rm = TRUE)
+```
+
+Grabbing summary statistics for columns of the whole dataset without needing to transfer data from Crunch servers is nice, but we're not going to get very far with just that. Say we want to see if there are differences in job satisfaction between people who never work from home and those who are full-time remote.
+
+We could pull down the `HomeRemote` column, and make a mask for rows that correspond to our three categories locally, then pull down the `JobSatsifaction` column, and find the mean of each of the groups.
+```{r}
+remote_col <- as.vector(so_data$HomeRemote)
+is_all_remote <- remote_col == "All or almost all the time (I'm full-time remote)"
+is_some_remote <- remote_col != "All or almost all the time (I'm full-time remote)" & remote_col != "Never"
+is_never_remote <- remote_col == "Never"
+
+job_satisfaction <- as.vector(so_data$JobSatisfaction)
+```
+```{r}
+mean(job_satisfaction[is_all_remote], na.rm=TRUE)
+mean(job_satisfaction[is_some_remote], na.rm=TRUE)
+mean(job_satisfaction[is_never_remote], na.rm=TRUE)
+```
+
+But as we saw before this means transferring at least two full columns worth of data, when Crunch has the ability to do all of this on the server with the use of filters. Using filters on Crunch datasets is easy and designed to approximate how we use filters on dataframes. Simply specify a logical expression (in this case it is a _Crunch logical expression_), and then we can use the same `mean` functions we would use normally. In this case, instead of pulling the entirety of each column from the server, instead the Crunch package sends the filter expression, the data is filtered on the server, and then the means are returned.
```

Comment:
I have mixed feelings, on balance positive, about expressing class names in english as `ClassName` ➡️  _Class name_